sdk/identity/Azure.Identity/src/MsalConfidentialClient.cs

```diff
@@ -146,7 +144,7 @@ protected override async ValueTask<IConfidentialClientApplication> CreateClientA
 
             if (!string.IsNullOrEmpty(tenantId))
             {
-                builder.WithAuthority(Pipeline.AuthorityHost.AbsoluteUri, tenantId);
+                builder.WithAuthority(AuthorityHost.AbsoluteUri, tenantId);
```

Comment:
>WithAuthority [](http://example.com/codeflow?start=24&length=13)

Could we use WithTenant here instead of WithAuthority? 

We don't intend or need to change the authority host -- that was set up when the ConfidentialClientApp was built.

Reply:
I believe there was some edge case scenario where this indirect approach to setting tenantId was what we implemented. I'd rather not change it as part of this PR to avoid possible regressions.